cellprofiler/modules/plugins/measurementtemplate.py

```diff
@@ -257,39 +358,38 @@ def run(self, workspace):
         #
         for n, m in self.get_zernike_indexes():
             # Compute the zernikes for each object, returned in an array
-            zr, zi = self.measure_zernike(
-                    pixels, labels, indexes, centers, radius, n, m)
+            zr, zi = self.measure_zernike(pixels, labels, indexes, centers, radius, n, m)
+
             # Get the name of the measurement feature for this zernike
             feature = self.get_measurement_name(n, m)
+
             # Add a measurement for this kind of object
             if m != 0:
-                meas.add_measurement(input_object_name, feature, zr)
-                #
+                measurements.add_measurement(input_object_name, feature, zr)
+
                 # Do the same with -m
-                #
                 feature = self.get_measurement_name(n, -m)
-                meas.add_measurement(input_object_name, feature, zi)
+                measurements.add_measurement(input_object_name, feature, zi)
             else:
                 # For zero, the total is the sum of real and imaginary parts
-                meas.add_measurement(input_object_name, feature, zr + zi)
-            #
+                measurements.add_measurement(input_object_name, feature, zr + zi)
```

Comment:
It looks like you're adding several values for a feature with the same name. Do you mind describing this behavior?

Reply:
Would you like me to clarify why `zr`, `zi` are arrays, and how that works with respect to saving per-object measurements in CellProfiler?

Reply:
I just realized that `feature` is changing for every measurement you add. Nope, this should be good :)